src/main/java/com/aws/greengrass/logmanager/util/PeriodicBuffer.java

```diff
@@ -0,0 +1,324 @@
+        }
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void performFlush(Map<K, V> bufferToFlush) {
+        logger.atInfo().log("Flushing buffer '{}' with {} items", bufferName, bufferToFlush.size());
+        long startTime = System.currentTimeMillis();
+        
+        try {
+            flushHandler.accept(bufferToFlush);
+            
+            long duration = System.currentTimeMillis() - startTime;
+            flushSuccessCount.incrementAndGet();
+            
+            logger.atDebug()
+                    .kv("itemCount", bufferToFlush.size())
+                    .kv("durationMs", duration)
+                    .kv("successCount", flushSuccessCount.get())
+                    .log("Successfully flushed buffer '{}'", bufferName);
+                    
+        } catch (RejectedExecutionException e) {
+            synchronized (bufferLock) {
+                // Restore items to buffer if flush failed
+                logger.atError().cause(e).log("Flush rejected for buffer '{}', items restored", bufferName);
+                bufferToFlush.forEach(buffer::putIfAbsent);
+            }
+            flushFailureCount.incrementAndGet();
+            logger.atError().cause(e)
+                    .kv("failureCount", flushFailureCount.get())
+                    .log("Flush rejected for buffer '{}', items restored", bufferName);
+        } catch (RuntimeException e) {
+            flushFailureCount.incrementAndGet();
+            logger.atError().cause(e)
+                    .kv("failureCount", flushFailureCount.get())
+                    .log("Flush failed for buffer '{}'", bufferName);
+        }
+    }
+
+    /**
+     * Returns the current size of the buffer.
+     */
+    public int size() {
+        synchronized (bufferLock) {
+            return buffer.size();
+        }
+    }
+
+    /**
+     * Checks if the buffer is empty.
+     */
+    public boolean isEmpty() {
+        synchronized (bufferLock) {
+            return buffer.isEmpty();
+        }
+    }
+
+    /**
+     * Shuts down the buffer, flushing any remaining items.
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public void shutdown() {
+        if (isShutdown) {
+            return;
+        }
+
+        isShutdown = true;
+        logger.atInfo().log("Shutting down buffer '{}'", bufferName);
+
+        // Cancel the scheduled task
+        if (flushTask != null) {
+            flushTask.cancel(false);
+        }
+        
+        // Then flush any remaining items
+        try {
+            flush(true);
+            currentFlushFuture.join(); // Wait for final flush to complete
+        } catch (RuntimeException e) {
+            logger.atWarn().cause(e).log("Exception during shutdown flush for buffer '{}'", bufferName);
+        }
+
+        // Shutdown the scheduler
+        scheduler.shutdown();
+        try {
+            if (!scheduler.awaitTermination(5, TimeUnit.SECONDS)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

These configs are flagged under GS-Tech Static config campaign and is considered to be a violation. Please migrate these configs